esmvaltool/diag_scripts/ipcc_ar5/ch09_fig09_17.py

```diff
@@ -0,0 +1,346 @@
+        if layer != '':
+            if cube_layer.coords('depth'):
+                z_units = cube_layer.coord('depth').units
+            else:
+                z_units = ''
+            title = ' '.join([title, '(', layer, str(z_units), ')'])
+        plt.title(title)
+        plt.legend(loc='best')
+        plt.ylabel(str(cube_layer.units))
+
+        # Determine image filename:
+        if multi_model:
+            path = diagtools.get_image_path(
+                cfg,
+                metadata,
+                prefix='MultiModel',
+                suffix='_'.join(['timeseries',
+                                 str(layer) + image_extention]),
+                metadata_id_list=[
+                    'field', 'short_name', 'preprocessor', 'diagnostic',
+                    'start_year', 'end_year'
+                ],
+            )
+
+        else:
+            path = diagtools.get_image_path(
+                cfg,
+                metadata,
+                suffix='timeseries_' + str(layer_index) + image_extention,
+            )
+
+        # Saving files:
+        if cfg['write_plots']:
+
+            logger.info('Saving plots to %s', path)
+            plt.savefig(path)
+
+        plt.close()
+
+
+def multi_model_time_series(
+        cfg,
+        metadata,
+        cubedic,
+):
+    """
+    Make a time series plot showing several preprocesssed datasets.
+
+    This tool loads several cubes from the files, checks that the units are
+    sensible BGC units, checks for layers, adjusts the titles accordingly,
+    determines the ultimate file name and format, then saves the image.
+
+    Parameters
+    ----------
+    cfg: dict
+        the opened global config dictionairy, passed by ESMValTool.
+    metadata: dict
+        The metadata dictionairy for a specific model.
+
+    """
+    # Load image format extention
+    image_extention = diagtools.get_image_format(cfg)
+
+    # Make the plot
+    plot_details = {}
+    cmap = plt.cm.get_cmap('viridis')
+
+    # Plot each file in the group
+    for index, filename in enumerate(sorted(cubedic)):
+        if len(metadata) > 1:
+            color = cmap(index / (len(metadata) - 1.))
+        else:
+            color = 'blue'
+
+        cube = cubedic[filename]
+
+        if 'MultiModel' in metadata[filename]['dataset']:
+            timeplot(
+                cube,
+                c=color,
+                # label=metadata[filename]['dataset'],
+                ls=':',
+                lw=2.,
+            )
+            plot_details[filename] = {
+                'c': color,
+                'ls': ':',
+                'lw': 2.,
+                'label': metadata[filename]['dataset']
+            }
+        else:
+            timeplot(
+                cube,
+                c=color,
+                # label=metadata[filename]['dataset'])
+                ls='-',
+                lw=2.,
+#                title = 'Ocean heat content anomaly [J]/n(with respect to 1971)',
```

Comment:
Please do not leave commented out code around

Reply:
yes, I will delete these lines and similar cases